### PR TITLE
Policy validation based on Policy Definition Action and Parameters

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
Policy validation based on Policy Definition Action and Parameters.
- Check policy action is supported by CC
- Check policy parameters
- Remove `order` attrib in policy
- Removed checking null Pointer exception in enforcer for attribute not found issues
- Remove HTTP protocol passing to OPA
- Add regex validation for param values

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2692
Fixes https://github.com/wso2/product-microgateway/issues/2660

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: Yes

### Tested environments

Tested with apictl

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/api_yaml.go [hunk 1]
```diff
@@ -110,11 +110,11 @@ type PolicyList []Policy
...
 type Policy struct {
-	PolicyName    string      `json:"policyName,omitempty"`
-	PolicyVersion string      `json:"policyVersion,omitempty"`
-	Action        string      `json:"-"`
-	Order         int         `json:"order,omitempty"`
-	Parameters    interface{} `json:"parameters,omitempty"`
+	PolicyName       string      `json:"policyName,omitempty"`
+	PolicyVersion    string      `json:"policyVersion,omitempty"`
+	Action           string      `json:"-"` // This is a meta value used in CC, not included in API YAML
+	IsPassToEnforcer bool        `json:"-"` // This is a meta value used in CC, not included in API YAML
```
Comment (pubudu538) [inline adapter/internal/oasparser/model/api_yaml.go:116]: What does this mean?

Comment (renuka-fernando) [reply adapter/internal/oasparser/model/api_yaml.go:116]: This `action` and `IsPassToEnforcer ` is not coming from the API YAML file (but others we get from it). We include these values in the policy for the internal process of the policy

File: adapter/internal/oasparser/model/policy_layout.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"errors"
+	"fmt"
+
+	"github.com/wso2/product-microgateway/adapter/internal/oasparser/constants"
+)
+
+// supportedPoliciesMap maps (policy action name) -> (policy layout)
+var supportedPoliciesMap = map[string]policyLayout{
+	"SET_HEADER": {
```
Comment (VirajSalaka) [inline adapter/internal/oasparser/model/policy_layout.go:29]: Constants here?

File: api/proto/wso2/health/service/health.proto [hunk 3]
```diff
@@ -24,7 +23,6 @@ message HealthCheckResponse {
 }
 
 // [#protodoc-title: Health]
-service Health { //TODO: (renuka) rename to HealthService ?
```
Comment (VirajSalaka) [inline api/proto/wso2/health/service/health.proto:27]: Keep the service name

**Review by VirajSalaka — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

File: adapter/internal/oasparser/model/policy_layout.go (hunk 2 above)
Comment (renuka-fernando) [reply adapter/internal/oasparser/model/policy_layout.go:29]: These are only used here in the Adapter. This action name is used only in Enforcer.

File: api/proto/wso2/health/service/health.proto (hunk 3 above)
Comment (renuka-fernando) [reply api/proto/wso2/health/service/health.proto:27]: Fixed with https://github.com/wso2/product-microgateway/pull/2805/commits/10af9235789ebfcd2591539f0815d15fe7f622a4. Thanks

File: adapter/internal/oasparser/model/async_api.go [hunk 4]
```diff
@@ -178,10 +178,9 @@ func populatePoliciesFromVendorExtensions(operation *Operation, vendorExtensions
 	policyParameters[constants.RewritePathResourcePath] = newResourcePath
 	policyParameters[constants.IncludeQueryParams] = true
 	policy := Policy{
-		PolicyName: constants.RewritePathResourcePath,
-		Action:     constants.RewritePathTemplate,
-		Order:      1,
-		Parameters: policyParameters,
+		Action:           constants.RewritePathAction,
```
Comment (suksw) [inline adapter/internal/oasparser/model/async_api.go:181]: Are we not reading the policy name from the vendor extensions?

File: adapter/internal/oasparser/model/async_api.go [hunk 5]
```diff
@@ -178,10 +178,9 @@ func populatePoliciesFromVendorExtensions(operation *Operation, vendorExtensions
...
+		Parameters:       policyParameters,
+		IsPassToEnforcer: true,
```
Comment (suksw) [inline adapter/internal/oasparser/model/async_api.go:183]: Isn't this used for other API types? other than AsyncAPIs?

File: adapter/internal/oasparser/model/api_yaml.go (hunk 1 above)
Comment (suksw) [reply adapter/internal/oasparser/model/api_yaml.go:116]: What would happen if `IsPassToEnforcer` was not set to true?

File: adapter/internal/oasparser/model/async_api.go (hunk 4 above)
Comment (renuka-fernando) [reply adapter/internal/oasparser/model/async_api.go:181]: Nope, this is not read from vendor extensions, Action is read from the policy definition go template file. But in here we have hard coded the policy "RewritePath" @VirajSalaka I am correct, right?

File: adapter/internal/oasparser/model/async_api.go (hunk 5 above)
Comment (renuka-fernando) [reply adapter/internal/oasparser/model/async_api.go]: `IsPassToEnforcer` is an internally used value, we are not getting this from API YAML or from outside, this is defined in the policy layout.
Fix hard coding true here by getting it from policy layouts with https://github.com/wso2/product-microgateway/pull/2805/commits/ff30a792424394e48e738a5927366d68925e70a8

**Review by suksw — APPROVED** [review]

File: adapter/internal/oasparser/model/api_yaml.go (hunk 1 above)
Comment (renuka-fernando) [reply adapter/internal/oasparser/model/api_yaml.go:116]: If it is not true, then the policy is not passed to the enforcer. That means the policy is not supported by the enforcer. For example, Call Interceptor Service API Policy do not want to be passed to the enforcer.

**Pull request merged**